Rework modbus (#28672)
home-assistant/home-assistant.github.io: source/_integrations/modbus.markdown

---
title: modbus

> description: Instructions on how to integrate modbus and platforms.

Reviewer: ```suggestion
description: Instructions on how to integrate modbus and platforms.
```
Is there a specific reason you changed the capitalization of Modbus? I'm usually all for sentence-style capitalization, but this seems to be a proper name. On their [website](https://modbus.org/), they use a capital letter.
[MS Style Guide](https://learn.microsoft.com/en-us/style-guide/grammar/nouns-pronouns) on proper names

Reviewer: I realize you're the Modbus expert here. Why change the spelling of Modbus to modbus? On their own website they always use a capital M.

ha_integration_type: integration
---

[modbus](http://www.modbus.org/) is a communication protocol to control PLCs (Programmable Logic Controller) and RTUs (Remote Terminal Unit).

> The integration adheres strictly to the [protocol specification](https://modbus.org/docs/modbus_Application_Protocol_V1_1b3.pdf) using [pymodbus](https://github.com/pymodbus-dev/pymodbus) for the actual protocol implmentation.

Reviewer: ```suggestion
The integration adheres strictly to the [protocol specification](https://modbus.org/docs/modbus_Application_Protocol_V1_1b3.pdf) using [pymodbus](https://github.com/pymodbus-dev/pymodbus) for the actual protocol implementation.
```

> The modbus integration supports all devices adhering to the modbus standard. The communication to the device(s) can be serial (rs-485), TCP, or UDP connections. The modbus integration allows multiple communication channels e.g. a serial port connection combined with one or more TCP connections.

Reviewer: ```suggestion
The modbus integration supports all devices adhering to the modbus standard. The communication to the devices can be serial (rs-485), TCP, or UDP connections. The modbus integration allows multiple communication channels. For example, a serial port connection combined with one or more TCP connections.
```

Author: the use of device(s) is to indicate that you can use one device or multiple devices,

e.g. is a correct abbreviation https://www.scribbr.com/definitions/for-example-abbreviation/#:~:text=The%20correct%20abbreviation%20for%20“for,like%20“ex%2Dboyfriend."

Reviewer: yes, I understand. And I realize that some Style Guides may allow it. But Home Assistant is using MS Style Guide which does not use [plural in parentheses](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/s-es).
Plural in parentheses does not work in some languages. The translator may have to make a decision whether to opt for plural or not. I know the HA website is not translated, but our UI is. And we're trying to be consistent.

Author: your suggestion is still basically wrong, but I changed it differently.

Author: is MS Style Guide is a demand for the web site (which is new!) then CI should check it.

# Configuring modbus communication

> Configure the modbus communication with modbus devices. This is a general setup needed establish access to the device.

Reviewer: ```suggestion
Configure the modbus communication with modbus devices. This is a general setup which is needed to establish access to the device.
```

> The modbus integration allows you to use multiple connections each with multiple sensors etc.

Reviewer: ```suggestion
The Modbus integration allows you to use multiple connections, each with multiple sensors, etc.
```

The modbus integration provides a number of parameters to help communicate with "difficult" devices, these parameters are independent of the type of communication.

{% configuration %}
close_comm_on_error:
  description: "Close connection when an error occurs.
  Some serial-rs485 adapters deliver garble when opened, this leads to a disconnect and a new connect, which can continue.
  If the log contains a message from pymodbus, with the text 'cleaning....', then try this parameter."
  required: false
  default: true
  type: boolean
delay:
  description: "Time to delay sending messages in seconds after connecting.
  Some modbus devices need a delay of typically 1-2 seconds after connection is established to prepare the communication.
  If a device does not respond to messages after connecting, then try this parameter.

>   **Remark:** the delay solely affect the first message."

Reviewer: ```suggestion
  **Remark:** the delay solely affects the first message."
```

Author: it is singular not plural.

  required: false
  default: 0
  type: integer
message_wait_milliseconds:
  description: "Time to wait in milliseconds between requests."
  required: false
  default: 30 for serial connection, 0 for all other connections.
  type: integer
name:
  description: "Name of this hub. Must be unique."
  required: true
  type: string
retries:
  description: "Number of times to retry a request."
  required: false
  default: 3
  type: integer
retry_on_empty:

>   description: "Retry request, when receiving and empty message."

Reviewer: ```suggestion
  description: "Retry request, when receiving an empty message."
```

  default: 5
  type: integer
type:
  description: "Type of modbus."
  required: true
  type: list
  keys:
    tcp:

>       description: "TCP/IP connection with socket framer, used with ethernet enabled devices."

Reviewer: ```suggestion
      description: "TCP/IP connection with socket framer, used with Ethernet enabled devices."
```

    udp:

>       description: "Udp connection with socket framer, rarely used."

Reviewer: ```suggestion
      description: "UDP connection with socket framer, rarely used."
```

    rtuovertcp:
      description: "TCP/IP connection with rtu framer, used when connection to modbus forwarders."
    serial:

>       description: "Serial connection with rtu framer, used with tty port or USB rs485 converter."

Reviewer: ```suggestion
      description: "Serial connection with RTU framer, used with TTY port or USB rs485 converter."
```

{% endconfiguration %}

## Configuring tcp connection

`type: tcp` is required. Used for devices providing a TCP/IP interface directly.

{% configuration %}
host:
  description: "IP address or name of your modbus device, e.g., `192.168.1.1`."
  required: true
  type: string
port:
  description: "Network port for the communication."
  required: true
  type: integer

{% endconfiguration %}

### Example: typical tcp configuration
```yaml
# Example yaml: typical tcp connection
modbus:
  - name: modbus_hub
    type: tcp
    host: IP_ADDRESS
    port: 502
```

### Example: full tcp configuration

```yaml
# Example yaml: full tcp connection
modbus:
  - name: modbus_hub
    type: tcp
    host: IP_ADDRESS
    port: 502

    close_comm_on_error: true
    delay: 0
    message_wait_milliseconds: 30
    retries: 3
    retry_on_empty: false
    timeout: 5
```

> ## Configuring tcp-rtu connection

Reviewer: ```suggestion
## Configuring a TCP-RTU connection
```

`type: rtuovertcp` is required. Used for devices providing a TCP/IP interface directly.

This is typically used, when communicating with a modbus-forwarder, a device that
has a TCP/IP connection upwards, and one or more serial connections downwards.

{% configuration %}
host:
  description: "IP address or name of your modbus device, e.g., `192.168.1.1`."
  required: true
  type: string
port:
  description: "Network port for the communication."
  required: true
  type: integer

{% endconfiguration %}

> ### Example: typical tcp-rtu configuration

Reviewer: ```suggestion
### Example: typical TCP-RTU configuration
```

```yaml
# Example yaml: typical tcp-rtu connection
modbus:
  - name: modbus_hub
    type: rtuovertcp
    host: IP_ADDRESS
    port: 502
```

> ### Example: full tcp-rtu configuration

Reviewer: ```suggestion
### Example: full TCP-RTU configuration
```

```yaml
# Example yaml: full tcp-rtu connection
modbus:
  - name: modbus_hub
    type: rtuovertcp
    host: IP_ADDRESS
    port: 502

    close_comm_on_error: true
    delay: 0
    message_wait_milliseconds: 30
    retries: 3
    retry_on_empty: false
    timeout: 5
```

> ## Configuring udp connection

Reviewer: ```suggestion
## Configuring a UDP connection
```

`type: udp` is required. This is rarely used, and only for very special configurations.

{% configuration %}
host:
  description: "IP address or name of your modbus device, e.g., `192.168.1.1`."
  required: true
  type: string
port:
  description: "Network port for the communication."
  required: true
  type: integer

{% endconfiguration %}

### Example: typical udp configuration
```yaml
# Example yaml: typical udp connection
modbus:
  - name: modbus_hub
    type: udp
    host: IP_ADDRESS
    port: 502
```

> ### Example: full udp configuration

Reviewer: ```suggestion
### Example: full UDP configuration
```

```yaml
# Example yaml: full udp connection
modbus:
  - name: modbus_hub
    type: udp
    host: IP_ADDRESS
    port: 502

    close_comm_on_error: true
    delay: 0
    message_wait_milliseconds: 30
    retries: 3
    retry_on_empty: false
    timeout: 5
```

## Configuring serial connection

`type: serial` is required. This is used for devices providing a serial rs485 interface.

The physical interface is typically a USB serial-rs485 converter or an rs232-rs485 connected to a serial port.

{% configuration %}
baudrate:
  description: "Speed of the serial connection, higher speed gives better performance."
  required: true
  type: integer
bytesize:
  description: "Data size in bits of each byte."
  required: true
  type: list
  keys:
    "5":
      description: "5 bit for data, rarely used."
    "6":
      description: "6 bit for data, rarely used."
    "7":
      description: "7 bit for data, used for very old devices."
    "8":
      description: "8 bit for data, standard."
method:
  description: "Method of the connection to modbus."
  required: true
  type: list
  keys:
    rtu:

>       description: "Binary data transmission preceded by slave id and followed by a crc, standard."

Reviewer: ```suggestion
      description: "Binary data transmission preceded by client id and followed by a CRC standard."
```
I could be wrong on this. But Modbus updated their terminology to use [client-server](https://modbus.org/docs/Client-ServerPR-07-2020-final.docx.pdf). Ignore this comment if there is a specific reason why we can't update to the client-server model.

Author: First reason is that HA connect to a server not a client, so your change is wrong. The other reason is that a server/client do not have an id, slave/unit does.

Reviewer: OK, thank you for explaining. If the client-server terminology change from Modbus cannot be applied here, please disregard my comments on the subject

slave:
  description: "Id of the device. Used to address multiple devices on a rs485 bus or devices connected to a modbus repeater."
  required: false
  type: integer
  default: 0
unique_id:
  description: "ID that uniquely identifies this entity.
  Slaves will be given a unique_id of <<unique_id>>_<<slave_index>>.
  If two enities have the same unique ID, Home Assistant will raise an exception."
  required: false
  type: string

{% endconfiguration %}

```yaml
# Example yaml: entities grouping
modbus:
  - type: tcp
    host: IP_ADDRESS_1
    port: 2020
    name: "modbus_hub"
    binary_sensors:
      - name: binary_sensor1
        address: 100
    climates:
      - name: "Watlow F4T"
        address: 200
    covers:
      - name: Door1
        address: 300
    fans:
      - name: Fan1
        address: 400
    lights:
      - name: light1
        address: 500
    sensors:
      - name: sensor1
        address: 600
    switches:
      - name: Switch1
        address: 700
```

The different types of entities are detailed in the following.

## Configuring binary sensor entities

The Modbus binary sensor allows you to gather data from coils which as per standard have state ON/OFF.

> Normally a register contains 16 coils, giving different address depending on the request used

Reviewer: ```suggestion
Normally, a register contains 16 coils, giving different addresses depending on the request used.
```

```yaml
Register 512: Coil 1 - 16
Register 513: Coil 17 - 32
```

> `input_type: coils` would use adresses from 1 through 32, while `input_type: input` would use adresses 512 and 513.

Reviewer: ```suggestion
`input_type: coils` would use addresses from 1 through 32, while `input_type: input` would use addresses 512 and 513.
```

> For that reason many devices (especially older ones) do not share the coil address space with the register address space,

Reviewer: ```suggestion
For that reason, many devices (especially older ones) do not share the coil address space with the register address space,
```

and this `input` would read from a different address space than `coil`. The problem is present in devices with
shared address space and are a frequent cause of problems when configuring entities.

{% configuration %}
binary_sensors:
  description: "A list of all binary sensors configured for this connection."
  required: false
  type: [map]
  keys:
    device_class:
      description: "The [type/class](/integrations/binary_sensor/#device-class) to be used for the UI."
      required: false
      type: string
    input_type:
      description: "Type of request `discrete_input`, `coil`, `holding` or `input`"
      required: false
      default: coil
      type: string
    slave_count:

>       description: "Generate count+1 binary sensors (master + slaves).

Reviewer: ```suggestion
      description: "Generate count+1 binary sensors (server + client).
```

Author: ???? We do not talk about different client, please read the text.

      Addresses are automatically incremented.
      The parameter simplifies configuration and provides a much better performance by not using count+1 requests but a single request."
      required: false
      type: integer
    unique_id:

>       description: "ID that uniquely identifies the entity. Slaves will automatically be given a unique_id of <<unique_id>>_<<slave_index>>.  If two sensors have the same unique ID, Home Assistant will raise an exception."

Reviewer: ```suggestion
      description: "ID that uniquely identifies the entity. Clients will automatically be given a unique_id of <<unique_id>>_<<client_index>>.  If two sensors have the same unique ID, Home Assistant will raise an exception."
```

Author: ??? we do not talk about different clients, please read the text.

      type: string

{% endconfiguration %}

### Example: typical binary sensor configuration

```yaml
# Example yaml: typical binary_sensor
modbus:
  - name: hub1
    type: tcp
    host: IP_ADDRESS
    port: 502
    binary_sensors:
      - name: my_relay
        address: 100
        slave: 1
```

### Example: full binary sensor configuration

```yaml
# Example yaml: binary_sensor with all options
modbus:
  - name: hub1
    type: tcp
    host: IP_ADDRESS
    port: 502
    binary_sensors:
      - name: my_relay
        address: 100
        device_class: door
        input_type: coil
        lazy_error_count: 0
        scan_interval: 15
        slave: 1
        slave_count: 0
        unique_id: my_relay
```

### Example: multiple identical binary sensor configuration

```yaml
# Example of 10 identical binary_sensor
modbus:
  - name: hub1
    type: tcp
    host: IP_ADDRESS
    port: 502
    binary_sensors:
      - name: my_relay
        address: 100
        slave: 1
        slave_count: 10
        unique_id: my_relay
```
This configuration will poll coil addresses 100 to 110 every 15 seconds and update the binary_sensors: `my_relay`
and `my_relay_1` to `my_relay_10`.

> The master configuration like device_class are automatically copied to the slaves.

Reviewer: ```suggestion
The master configuration like device_class are automatically copied to the clients.
```

Author: ??? We do not talk about different clients, please read the text.

        int64:
          description: "64 bit signed integer (4 registers holds 1 value)."
        string:
          description: "set of 8 bit characters, `count:` must be configured."
        uint:
          description: "**DEPRECATED** is silently converted to `uint16`"
        uint8:
          description: "8 bit unsigned integer (1 register holds 1 value in lower byte)."
        uint16:
          description: "16 bit unsigned integer (1 register holds 1 value)."
        uint32:
          description: "32 bit unsigned integer (2 registers holds 1 value)."
        uint64:
          description: "64 bit unsigned integer (4 registers holds 1 value)."
    hvac_mode_register:
      description: "Configuration of register for HVAC mode"
      required: false
      type: [map]
      keys:
        address:
          description: "Address of HVAC mode register."
          required: true
          type: integer
        write_registers:
          description: "Request type, use `write_registers` if true  else `write_register`."
          required: false
          type: boolean
          default: false
        values:

>           description: "Mapping between the register values and HVAC modes"

Reviewer: ```suggestion
          description: "Mapping between the registered values and HVAC modes."
```

Author: No it is the "register value" as in "value of the register" the registred value is something different.

Reviewer: Ah, I see. Thank you for explaining.

    type: tcp
    host: IP_ADDRESS
    port: 502
    climates:
      - name: "Watlow F4T"
        address: 27586
        input_type: holding
        count: 1
        data_type: custom
        max_temp: 35
        min_temp: 15
        offset: 0
        precision: 1
        scale: 0.1
        max_temp: 30
        structure: ">f"
        target_temp_register: 2782
        temp_step: 1
        temperature_unit: C
```

## Configuring cover entities

The `modbus` cover platform allows you to control covers (such as blinds, a roller shutter, or a garage door).

At the moment, platform cover support the opening and closing of a cover. You can control your covers either using coils or holding registers.

> Cover that uses `input_type: coil` is not able to determine intermediary states such as opening and closing. Coil stores only two states — "0" means cover closed, and "1" implies cover open. To allow detecting intermediary states, there is an optional `status_register` attribute. It will enable you to write your command (e.g., to open a cover) into a coil, and read current cover status back through the register. Additionally, you can specify values for `state_open`, `state_opening`, `state_closed`, and `state_closing` attributes. These will be matched with the value read from the `status_register`.

Reviewer: ```suggestion
Cover that uses `input_type: coil` is not able to determine intermediary states such as opening and closing. Coil stores only two states — "0" means cover closed, and "1" implies cover open. To allow detecting intermediary states, there is an optional `status_register` attribute. It will enable you to write your command (for example to open a cover) into a coil, and read current cover status back through the register. Additionally, you can specify values for `state_open`, `state_opening`, `state_closed`, and `state_closing` attributes. These will be matched with the value read from the `status_register`.
```

    input_type:
      description: "Cover register type."
      default: holding
      required: false
      type: map
      keys:
        holding:
          description: "Holding register."
        input:
          description: "Input register."
    state_open:
      description: "A value in `status_register` or `register` representing an open cover.
        If your configuration uses the `register` attribute, this value will be written into the holding register to open the cover."
      required: false
      default: 1
      type: integer
    state_closed:
      description: "A value in `status_register` or `register` representing a closed cover.
        If your configuration uses the `register` attribute, this value will be written into the holding register to close the cover."
      required: false
      default: 0
      type: integer
    state_opening:
      description: "A value in `status_register` or `register` representing an opening cover.
        Note that this state should be also supported on your connected Modbus cover.
        If it won't report the state, this state won't be detected."
      required: false
      default: 2
      type: integer
    state_closing:

>       description: "A value in `status_register` or `register` representing a closing cov

Reviewer: ```suggestion
      description: "A value in `status_register` or `register` representing a closing cover.
```

        Note that this state should be also supported on your connected Modbus cover.

>         If it won't reeport the state, this state won't be detected."

Reviewer: ```suggestion
        If it doesn't report the state, this state won't be detected."
```

        slave: 1
        device_class: door
        scan_interval: 10
        register: 1
        status_register: 2
        register_type: holding
        state_open: 1
        state_closed: 0
```

## Configuring fan entities

The `modbus` fan platform allows you to control [Modbus](http://www.modbus.org/) coils or registers.

{% configuration %}
fans:
  description: "A list of all fan entities in this modbus instance."
  required: true
  type: map
  keys:
    command_on:
      description: "Value to write to turn on the fan."
      required: false
      default: 0x01
      type: integer
    command_off:
      description: "Value to write to turn off the fan."
      required: false
      default: 0x00
      type: integer
    write_type:
      description: Type of write request.
      required: false
      default: holding
      type: map
      keys:
        holding:
          description: "write_register is called."
        holdings:
          description: "write_registers is called."
        coil:
          description: "write_coil is called."
        coils:
          description: "write_coils is called."
    verify:
      description: "Read from Modbus device to verify fan.

>         If used without attributes it uses the toggle register configuration.

Reviewer: ```suggestion
        If used without attributes, it uses the toggle register configuration.
```

>         If omitted no verification is done, but the state of the fan is set with each toggle."

Reviewer: ```suggestion
        If omitted, no verification is done, but the state of the fan is set with each toggle."
```

            state_off: 1
```

## Configuring light entities

The `modbus` light platform allows you to control [Modbus](http://www.modbus.org/) coils or registers.

{% configuration %}
lights:
  description: "A list of all light entities in this modbus instance."
  required: true
  type: map
  keys:
    command_on:
      description: "Value to write to turn on the light."
      required: false
      default: 0x01
      type: integer
    command_off:
      description: "Value to write to turn off the light."
      required: false
      default: 0x00
      type: integer
    write_type:
      description: "Type of write request."
      required: false
      default: holding
      type: map
      keys:
        holding:
          description: "write_register is called."
        holdings:
          description: "write_registers is called."
        coil:
          description: "write_coil is called."
        coils:
          description: "write_coils is called."
    verify:
      description: "Read from Modbus device to verify the light.

>         If used without attributes it uses the toggle register configuration.

Reviewer: ```suggestion
        If used without attributes, it uses the toggle register configuration.
```

>         If omitted no verification is done, but the state of the light is set with each toggle."

Reviewer: ```suggestion
        If omitted, no verification is done, but the state of the light is set with each toggle."
```

      required: false
      type: device_class
      default: None
    input_type:
      description: "Modbus register type for sensor."
      required: false
      default: holding
      type: map
      keys:
        holding:
          description: "Holding register."
        input:
          description: "Input register."
    min_value:
      description: "The minimum allowed value of a sensor. If value < min_value --> min_value. Can be float or integer"
      required: false
      type: float
    max_value:
      description: "The maximum allowed value of a sensor. If value > max_value --> max_value. Can be float or integer"
      required: false
      type: float
    offset:
      description: "Final offset (output = scale * value + offset)."
      required: false
      type: float
      default: 0
    precision:
      description: "Number of valid decimals."
      required: false
      type: integer
      default: 0
    scale:
      description: "Scale factor (output = scale * value + offset)."
      required: false
      type: float
      default: 1
    slave_count:

>       description: "Generates x+1 sensors (master + slaves), allowing read of multiple registers with a single read messsage."

Reviewer: ```suggestion
      description: "Generates x+1 sensors (server + client), allowing read of multiple registers with a single read messsage."
```

Author: that is not clients !!

{% configuration %}
switches:
  description: "A list of all switches in this modbus instance."
  required: true
  type: map
  keys:
    command_on:
      description: "Value to write to turn on the switch."
      required: false
      default: 0x01
      type: integer
    command_off:
      description: "Value to write to turn off the switch."
      required: false
      default: 0x00
      type: integer
    write_type:
      description: Type of write request.
      required: false
      default: holding
      type: map
      keys:
        holding:
          description: "write_register is called."
        holdings:
          description: "write_registers is called."
        coil:
          description: "write_coil is called."
        coils:
          description: "write_coils is called."
    verify:
      description: "Read from Modbus device to verify fan.

>         If used without attributes it uses the toggle register configuration.

Reviewer: ```suggestion
        If used without attributes, it uses the toggle register configuration.
```

>         If omitted no verification is done, but the state of the fan is set with each toggle."

Reviewer: ```suggestion
        If omitted, no verification is done, but the state of the fan is set with each toggle."
```

          type: integer
        state_off:
          description: "Value when switch is off."
          required: false
          default: "Same as `command_off`"
          type: integer

{% endconfiguration %}

### Example: switch configuration

```yaml
# Example configuration.yaml entry
modbus:
  - type: tcp
    host: IP_ADDRESS
    port: 502
    switches:
      - name: Switch1
        address: 13
        write_type: coil
      - name: Switch2
        slave: 2
        address: 14
        write_type: coil
        verify:
      - name: Register1
        address: 11
        command_on: 1
        command_off: 0
        verify:
            input_type: holding
            address: 127
            state_on: 25
            state_off: 1
```

# modbus services

The modbus integration provides two generic write services in addition to the platform-specific services.

| Service | Description |
| ------- | ----------- |
| modbus.write_register | Write register or registers |
| modbus.write_coil | Write coil or coils |

Description:

| Attribute | Description |
| --------- | ----------- |
| hub       | Hub name (defaults to 'modbus_hub' when omitted) |

> | unit      | Slave address (0-255), alternative to slave |

Reviewer: ```suggestion
| unit      | Client address (0-255), alternative to client |
```

Author: No there are no client address in Modbus, only unit or slave id´s

> | slave     | Slave address (0-255), alternative to unit |

Reviewer: ```suggestion
| slave     | Client address (0-255), alternative to unit |
```

Author: Just to make it clear, HA is a client that connects to a server, so your change is wrong....apart from the fact that slave/unit is the Modbus terminology also used in configuration.

| address   | Address of the Register (e.g. 138) |

> | value     | (write_register) A single value or an array of 16-bit values. Single value will call modbus function code 0x06. Array will call modbus function code 0x10. Values might need reverse ordering. E.g., to set 0x0004 you might need to set `[4,0]`, this depend on the byte order of your CPU |

Reviewer: ```suggestion
| value     | (write_register) A single value or an array of 16-bit values. Single value will call modbus function code 0x06. Array will call modbus function code 0x10. Values might need reverse ordering. For example, to set 0x0004 you might need to set `[4,0]`. This depends on the byte order of your CPU |
```

Author: "this" relates to the example making it an independent sentence changes that.

| state     | (write_coil) A single boolean or an array of booleans. Single boolean will call modbus function code 0x05. Array will call modbus function code 0x0F |

The modbus integration also provides communication stop/restart services. These services will not do any reconfiguring, but simply stop/start the modbus communication layer.

| Service | Description |
| ------- | ----------- |
| modbus.stop | Stop communication |
| modbus.restart | Restart communication (Stop first if running) |

Description:

| Attribute | Description |
| --------- | ----------- |
| hub       | Hub name (defaults to 'modbus_hub' when omitted) |

## Example: writing a float32 type register

> To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal).

Reviewer: ```suggestion
To write a float32 datatype register, use a network format like `10.0` == `0x41200000` (network order float hexadecimal).
```

Author: "a" indicated "one of of several" which is not the case, there are only 1 network format.

```yaml
service: modbus.write_register
data:
  address: <target register address>
  unit: <target slave address>
  hub: <hub name>
  value: [0x4120, 0x0000]
```

## Service `modbus.set-temperature`

| Service | Description |
| ------- | ----------- |

> | set_temperature | Set Temperature. Requires `value` to be passed in, which is the desired target temperature. `value` should be in the same type as `data_type` |

Reviewer: ```suggestion
| set_temperature | Set temperature. Requires `value` to be passed in, which is the desired target temperature. `value` should be in the same type as `data_type` |
```
